### PR TITLE
Ermish/vue3refresh

### LINKED ISSUE #30: Upgrade to Vue3 and Nuxt 3

### PR DESCRIPTION
**Pull request by ermish**
# Site is alive with vue3/nuxt3!
Resolves #30, #23, #24, #25
- Upgraded to Vue3
- Upgraded to Nuxt3
- Upgraded to Yarn 3.2.2 (berry)
- Remove imcompatible google analytics nuxt plugin (for now)
- Created Home/Search page, about, news, login, admin dashboard empty templates.
- migrated components to vue3 
- added pinia for state management
- Adding in eslint, stylelint, editorconfig
- updated tsconfig to support vue3
- Updated yarn build commands 
- Added min node version dependency for builds

### CODE REVIEW

**Comment by theyokohamalife** [comment]
@ermish I tried to run this using `yarn dev` but I'm getting the following error. It looks like it's being caused by Nitro?
Screen Shot 2022-07-21 at 11 59 39 PM

**Comment by theyokohamalife** [comment]
It seems that the `Vue.extend()` you're using in the LoginPage and AdminDashboardPage are causing the issue. I removed those and was finally able to get the server to run. Is this block of code needed?

```
<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
    name: 'IndexPage'
})
</script>
```

**Comment by theyokohamalife** [comment]
@ermish I'd like to see a preview of this on Netlify before it gets merged to main. Can you re-enable previews for PRs?

**Comment by ermish** [comment]
oh wierd! I updated that 🤔  perhaps I have a local commit I didn't check in


> It seems that the `Vue.extend()` you're using in the LoginPage and AdminDashboardPage are causing the issue. I removed those and was finally able to get the server to run. Is this block of code needed?
> 
> ```
> <script lang="ts">
> import Vue from 'vue'
> 
> export default Vue.extend({
>     name: 'IndexPage'
> })
> </script>
> ```

**Comment by ermish** [comment]
> @ermish I'd like to see a preview of this on Netlify before it gets merged to main. Can you re-enable previews for PRs?

finally got them working! Had to upgrade the netlify build image and node version dependency. 
Here's the new preview
https://deploy-preview-29--findadoc.netlify.app/

**Comment by theyokohamalife** [comment]
@ermish Had a look at the preview and noticed that there are links to the empty pages in the top app bar. Can those be removed until they're ready and we've decided on a navigation system? Also, I think we can remove the News component altogether, since it won't be included in the MVP.

**Comment by theyokohamalife** [comment]
@ermish I'm still unable to run the project. I'm now getting the following message after `yarn install` and `yarn dev`: 
Screen Shot 2022-07-24 at 3 49 24 PM

**Comment by ermish** [comment]
> @ermish I'm still unable to run the project. I'm now getting the following message after `yarn install` and `yarn dev`: 
> 
> Screen Shot 2022-07-24 at 3 49 24 PM
> 
> 

Hmmm not sure why. That seems to be a package issue 🤔 we might need to walk through this one

**Comment by ermish** [comment]
> @ermish Had a look at the preview and noticed that there are links to the empty pages in the top app bar. Can those be removed until they're ready and we've decided on a navigation system? 
> 
> 
> 
> Screen Shot 2022-07-24 at 3 38 15 PM
> 
> 

For now, I just want to get this in here so I can start developing again. Since these aren't going out to production I didn't want to add temporary logic for hiding the navbar. There will be a whole separate ticket for working on the navbar section and the welcome banner. I just wanted to verify the vue3/nuxt3 routing is working and have an example

**Comment by ermish** [comment]
@theyokohamalife can we get this merged in and I'll investigate the build in the next PR?

**Comment by theyokohamalife** [comment]
> @theyokohamalife can we get this merged in and I'll investigate the build in the next PR?

I understand that you want to get this merged quickly, but also please understand that I can't approve it if I can't get it working.

It might help if you break this PR down into smaller ones, because it's trying to do a lot of things and it's hard to pinpoint where things are going wrong.

File: .eslintrc.js
Comment (ann-kilzer) [inline .eslintrc.js:16]: Might it be better for us to adopt a set of rules (like airbnb) that we can use across our different repos, rather than hand-configuring different rules?

File: pages/newsPage.vue
Comment (ann-kilzer) [inline pages/newsPage.vue:1]: I think we should skip this one for MVP

File: components/Welcome.vue
Comment (ann-kilzer) [inline components/Welcome.vue:14]: Can we set the linter to check line width? This line is too long

File: .eslintrc.js
Comment (ermish) [reply .eslintrc.js:16]: For now, I just used my custom set of rules to get started. IMO there are some gaps in most of the default packages such as Airbnb.
We should walk through linting rules in our next dev meeting 👍

Comment (ermish) [reply .eslintrc.js:16]: After we discuss, I can update them in a future commit

File: components/Welcome.vue
Comment (ermish) [reply components/Welcome.vue:14]: Yeah I haven't run the linter across the project or adding linting automation yet 👍 will fix in future linter setup commit #42

File: pages/newsPage.vue
Comment (ermish) [reply pages/newsPage.vue:1]: I just added it as an empty page to test out the navigation. 
I'll remove it

File: pages/admindashboardPage.vue
Comment (ermish) [reply pages/admindashboardPage.vue:1]: Good find, updating.

**Comment by ermish** [comment]
@ann-kilzer updated! please check  :)

**Comment by ermish** [comment]
@theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now. 

@theyokohamalife @ann-kilzer 
To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`

**Comment by theyokohamalife** [comment]
> @theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now.
> 
> @theyokohamalife @ann-kilzer To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`

@ermish I'm using yarn `3.2.2`

**Comment by ermish** [comment]
> > @theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now.
> > @theyokohamalife @ann-kilzer To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`
> 
> @ermish I'm using yarn `3.2.2`

@theyokohamalife ok we should be good to merge! Please check if it works for you now

File: .editorconfig
Comment (theyokohamalife) [inline .editorconfig:8]: I'd prefer to remove the indentation changes now instead of merging them into main and having to undo them.

File: .eslintrc.js
Comment (theyokohamalife) [inline .eslintrc.js:48]: Same here and other places.

File: .stylelintrc.json
Comment (theyokohamalife) [inline .stylelintrc.json:9]: Here as well. I'm probably missing some lines.

File: components/TopNav.vue
Comment (theyokohamalife) [inline components/TopNav.vue:32]: I don't want to check these `<script>` sections in for a few reasons.

1. They aren't currently being used and are just adding unnecessary lines to the code.
2. They might encourage people to start importing and exporting components the Vue way instead of the Nuxt way, which defeats one of the biggest reasons for using Nuxt.

For now, please remove these. 🙏🏾

File: pages/aboutPage.vue
Comment (theyokohamalife) [inline pages/aboutPage.vue:19]:
```suggestion

```

File: pages/adminDashboardPage.vue
Comment (theyokohamalife) [inline pages/adminDashboardPage.vue:15]:
```suggestion

```

File: pages/index.vue
Comment (theyokohamalife) [inline pages/index.vue:29]: Please remove 🙏🏾

File: pages/loginPage.vue
Comment (theyokohamalife) [inline pages/loginPage.vue:14]:
```suggestion

```

**Review by theyokohamalife — CHANGES_REQUESTED** [review]

File: components/TopNav.vue
Comment (theyokohamalife) [inline components/TopNav.vue:7]:
```suggestion
           
```

Comment (theyokohamalife) [reply components/TopNav.vue:7]: This link should be remove since the `newsPage` has been deleted.

File: .editorconfig
Comment (ermish) [reply .editorconfig:8]: at this point it will be unnecessary work. Most of these files are new and created in 4, so there's no undo-ing.
 I'm hoping to make the case for 4 instead of 2. :)

File: components/TopNav.vue
Comment (ermish) [reply components/TopNav.vue:32]: these files are literally for my personal use at the moment and I'm going to be filling out a lot of logic in these script sections. For now these are placeholders that I'm actively filling out locally until this can get merged in. 

This PR is just templating, not final files.

File: pages/aboutPage.vue
Comment (ermish) [reply pages/aboutPage.vue:19]: Is there a difference in the suggestion? 😅  I can't tell

Comment (ermish) [reply pages/aboutPage.vue:19]: ohhhh...it's to remove it....I'm literally re-adding them in in future commits. For now, we should leave them. Especially considering I'm the sole developer on these for now. Before we have other contributors work on them, the files will be completely different

File: components/TopNav.vue
Comment (ermish) [reply components/TopNav.vue]: good find, updated

**Review by theyokohamalife — APPROVED** [review]

**Pull request merged**
